engines/scc_proxy/lib/scc_proxy/engine.rb

```diff
@@ -263,11 +272,33 @@ def has_no_regcode?(auth_header)
 
         protected
 
+        # rubocop:disable Metrics/PerceivedComplexity
+        # rubocop:disable Metrics/CyclomaticComplexity
         def scc_activate_product
           logger.info "Activating product #{@product.product_string} to SCC"
-          auth = request.headers['HTTP_AUTHORIZATION']
-          if @system.byos?
-            response = SccProxy.scc_activate_product(@product, auth, params[:token], params[:email])
+          auth = nil
+          auth = request.headers['HTTP_AUTHORIZATION'] if request.headers.include?('HTTP_AUTHORIZATION')
+          if @system.byos? || (!@system.byos? && !@product.free? && @product.extension?)
+            mode = 'byos' if @system.byos?
+
+            unless @system.byos?
+              # system is not BYOS and
+              # product is a non free extension
+              mode = 'hybrid'
+              # the extensions must be the same version and arch
+              # than base product
+              base_prod = @system.products.find_by(product_type: :base)
+              if base_prod.present? && @product.arch == base_prod.arch && @product.version == base_prod.version
+                request.headers['proxy_byos_mode'] = mode
+                request.headers['scc_login'] = @system.login
+                request.headers['scc_password'] = @system.password
+                request.headers['instance_data'] = params[:instance_data]
+                request.headers['hwinfo'] = params[:hwinfo]
+                response = SccProxy.announce_system_scc(auth, request.headers)
+              end
+            end
```

Comment:
Let's reduce the complexity instead of ignoring the check. Consider:

```
mode = nil
if @system.byos?
  mode = 'byos'
elsif !@product.free? && @product.extension?
  mode = 'hybrid'
  ...
end

Reply:
Yep the entry condition is overly complex.